lib/mayaHydra/mayaPlugin/renderOverride.cpp

```diff
@@ -155,6 +156,20 @@ PXR_NAMESPACE_USING_DIRECTIVE
 
 const SdfPath MAYA_NATIVE_ROOT = SdfPath("/MayaHydraViewportRenderer");
 
+TfToken _GetPurposeRenderTagFromAttrName(const TfToken& attrName)
+{
+    if (attrName == TfToken("mayaHydraRenderPurpose")) {
+        return HdRenderTagTokens->render;
+    } else if (attrName == TfToken("mayaHydraProxyPurpose")) {
+        return HdRenderTagTokens->proxy;
+    } else if (attrName == TfToken("mayaHydraGuidePurpose")) {
+        return HdRenderTagTokens->guide;
+    } else {
+        TF_CODING_ERROR("Unknown purpose attribute name '%s'", attrName.GetText());
+        return {};
+    }
```

Comment:
Style suggestion, do if you feel like it.
```suggestion
    {
        static std::map<TfToken, TfToken> attrToTag{
            {TfToken("mayaHydraRenderPurpose"), HdRenderTagTokens->render},
            {TfToken("mayaHydraProxyPurpose"),  HdRenderTagTokens->proxy},
            {TfToken("mayaHydraGuidePurpose"),  HdRenderTagTokens->guide}};
        auto found = attrToTag.find(attrName);
        if (found != attrToTag.end()) {
            return found->second;
        }
        TF_CODING_ERROR("Unknown purpose attribute name '%s'", attrName.GetText());
        return {};
    }
```